### PR TITLE
[ML] [Pipelines] refactor: move code related logic to a mixin

### PR DESCRIPTION
**Pull request by elliotzh**
# Description

This pull request is a preparation for a unified solution for attributes maintenance of `Component`. `code` is the only attribute involving complicated local build so isolate related logic from `Component` first.

1. Use the same temp cache folder for both `cache_utils` and `artifact_utils`.
2. Move most additional includes implementation from `Component` to `AdditionalIncludesMixin`.
3. Update property name, function name and related docstring for `AdditionalIncludes` and `InternalAdditionalIncludes`.
4. `InternalAdditionalIncludes` now have no direct usage but kept for compatibility with `mldesigner`. Difference between additional includes in `InternalComponent` and `CommandComponent` is now implemented inside `InternalComponent`.
5. Code resolution related logic has been moved to `ComponentCodeMixin` and base class for components doesn't have code related logic anymore.
6. Validation & resolution logic in `AdditionalIncludes` have been split to avoid caching a `AdditionalIncludes` object. Note that content conflict check will be done for artifact folder name only.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/ml/azure-ai-ml/azure/ai/ml/_internal/entities/_additional_includes.py [hunk 1]
```diff
@@ -46,5 +57,5 @@ def is_artifact_includes(self):
                     isinstance(additional_includes_configs, dict)
                     and self.get_config_key() in additional_includes_configs
                 )
-        except Exception:  # pylint: disable=broad-except
+        except (FileNotFoundError, yaml.YAMLError):
```
Comment (brynn-code) [inline sdk/ml/azure-ai-ml/azure/ai/ml/_internal/entities/_additional_includes.py]: Is it possible for any other exception here? If so, the exception will be raised directly, is it?

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_component/additional_includes.py [hunk 2]
```diff
@@ -127,6 +123,7 @@ def _get_artifacts_by_config(cls, artifact_config):
 
     def _resolve_additional_include_config(self, additional_include_config):
```
[the remark this answers is not in the document]
Comment (elliotzh) [reply sdk/ml/azure-ai-ml/azure/ai/ml/entities/_component/additional_includes.py]: Done the rename first and I'll try to reorganize related functions. hard to understand what's the difference between `_resolve_xxx` and `_load_xxx` from function name

File: sdk/ml/azure-ai-ml/azure/ai/ml/_internal/entities/_additional_includes.py (hunk 1 above)
Comment (elliotzh) [reply sdk/ml/azure-ai-ml/azure/ai/ml/_internal/entities/_additional_includes.py]: Updated

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_component/command_component.py [hunk 3]
```diff
@@ -265,46 +260,3 @@ def __str__(self):
             return self._to_yaml()
         except BaseException:  # pylint: disable=broad-except
             return super(CommandComponent, self).__str__()
-
-    @contextmanager
-    def _resolve_local_code(self) -> Optional[Code]:
```
Comment (wangchao1230) [inline sdk/ml/azure-ai-ml/azure/ai/ml/entities/_component/command_component.py:270]: kudos! code much cleaner now

File: sdk/ml/azure-ai-ml/azure/ai/ml/_internal/entities/_additional_includes.py [hunk 4]
```diff
@@ -2,52 +2,12 @@
...
 
-from pathlib import Path
-import yaml
-
-from ...entities._component._additional_includes import (
-    AdditionalIncludes,
-    ADDITIONAL_INCLUDES_KEY,
-    ADDITIONAL_INCLUDES_SUFFIX,
-)
+from ...entities._component._additional_includes import AdditionalIncludes
```
Comment (wangchao1230) [inline sdk/ml/azure-ai-ml/azure/ai/ml/_internal/entities/_additional_includes.py:5]: May take a look at the failed tests

**Review by wangchao1230 — APPROVED** [review]

**Pull request merged**